Codecademy/gamut
# feat(Modal + Dialog): add image prop to modal

### Overview


Adds the new `image` prop to Modal + Dialog , which renders the `image` modal when provided. 


### PR Checklist

- [x] Related to designs: [here](https://www.figma.com/file/ReGfRNillGABAj5SlITalN/%F0%9F%93%90-Gamut?type=design&node-id=2449%3A3770&t=d1ZWMWMqZdP2CGwt-1)
- [x] Related to JIRA ticket: [web-2311]

#### Testing Instructions



* Check out the image Modal examples in Storybook
* Check out the image Dialog examples in Storybook

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/35590)  | [Monolith Env](http://www.google.fr/ 'Named link title') |
| Portal       | [Portal PR](https://github.com/codecademy-engineering/mono/pull/1916)  | [Portal Env](http://www.google.fr/ 'Named link title')   |
